src/ir.h

```diff
@@ -578,6 +579,16 @@ class ReturnCallIndirectExpr : public ExprMixin<ExprType::ReturnCallIndirect> {
   Var table;
 };
 
+class CallRefExpr : public ExprMixin<ExprType::CallRef> {
+ public:
+  explicit CallRefExpr(const Location &loc = Location())
+      : ExprMixin<ExprType::CallRef>(loc) {}
+
+  // This field is setup only during Validate phase,
+  // so keep that in mind when you use it.
+  Var function_type_index;
```

Comment:
So there is no immediate in the binary?  Its just implied?

Reply:
Yes, according to this https://github.com/WebAssembly/function-references/blob/master/proposals/function-references/Overview.md#instructions-1 `call_ref` has no immediates, so we have to determine it's type from the context. 

I could use optional type here to express possible empty case more cleary but it looks like `Var` already has the value for indicating the invalid state.